tools/base/fpgainfo/bmcdata.c

```diff
@@ -111,24 +112,36 @@ fpga_result get_metrics(fpga_token token,
 }
 
 void print_metrics(const fpga_metric_info *metrics_info,
+                    uint64_t num_metrics_info,
                    const fpga_metric *metrics, uint64_t num_metrics)
 {
-    uint64_t i = 0;
-    for (i = 0; i < num_metrics; ++i) {
-        uint64_t idx = metrics[i].metric_num;
-        
-        if (idx < num_metrics) {
-            printf("%-29s : ", metrics_info[idx].metric_name);
-
-            if (metrics_info[i].metric_datatype == FPGA_METRIC_DATATYPE_INT) {
-                printf("%" PRId64 "", metrics[i].value.ivalue);
-            } else if (metrics_info[i].metric_datatype == FPGA_METRIC_DATATYPE_FLOAT ||
-                    metrics_info[i].metric_datatype == FPGA_METRIC_DATATYPE_DOUBLE) {
-                printf("%0.2f", metrics[i].value.dvalue);
-            }
+	uint64_t i = 0;
+	for (i = 0; i < num_metrics; ++i) {
+		uint64_t idx = metrics[i].metric_num;
 
-            printf(" %s\n", metrics_info[idx].metric_units);
-        }
-    }
+		if (metrics[i].isvalid) {
+
+			if (idx < num_metrics) {
+				printf("(%2ld)  %-29s : ", i + 1, metrics_info[idx].metric_name);
+
+				if (metrics_info[idx].metric_datatype == FPGA_METRIC_DATATYPE_INT) {
+					printf("%" PRId64 "", metrics[i].value.ivalue);
+				}
+				else if (metrics_info[idx].metric_datatype == FPGA_METRIC_DATATYPE_FLOAT ||
+					metrics_info[idx].metric_datatype == FPGA_METRIC_DATATYPE_DOUBLE) {
+					printf("%0.2f", metrics[i].value.dvalue);
+				}
+
```

Comment:
what about `FPGA_METRIC_DATATYPE_BOOL`?
Should this be a switch statement instead of if/else?
```C
switch (metrics_info[idx].metric_datatype) {
    case FPGA_METRIC_DATATYPE_INT:
        // ..
        break;
    case FPGA_METRIC_DATATYPE_DOUBLE: /* fallthrough */
    case FPGA_METRIC_DATATYPE_FLOAT:
        // ...
        break;
    case FPGA_METRIC_DATATYPE_BOOL:
        // ...
        break;
    default: // error
}
```

Reply:
fixed